pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/minion/PinotHelixTaskResourceManager.java

```diff
@@ -64,46 +81,65 @@ public void createTaskQueue(@Nonnull String taskType) {
   }
 
   /**
-   * Stop a task queue of the given task type.
+   * Stop the task queue for the given task type.
    *
    * @param taskType Task type
    */
-  public void stopTaskQueue(@Nonnull String taskType) {
+  public synchronized void stopTaskQueue(@Nonnull String taskType) {
     String helixJobQueueName = getHelixJobQueueName(taskType);
     LOGGER.info("Stopping task queue: {} for task type: {}", helixJobQueueName, taskType);
     _taskDriver.stop(helixJobQueueName);
   }
 
   /**
-   * Resume a task queue of the given task type.
+   * Resume the task queue for the given task type.
    *
    * @param taskType Task type
    */
-  public void resumeTaskQueue(@Nonnull String taskType) {
+  public synchronized void resumeTaskQueue(@Nonnull String taskType) {
     String helixJobQueueName = getHelixJobQueueName(taskType);
     LOGGER.info("Resuming task queue: {} for task type: {}", helixJobQueueName, taskType);
     _taskDriver.resume(helixJobQueueName);
   }
 
   /**
-   * Delete a task queue of the given task type.
+   * Delete the task queue for the given task type.
    *
    * @param taskType Task type
    */
-  public void deleteTaskQueue(@Nonnull String taskType) {
+  public synchronized void deleteTaskQueue(@Nonnull String taskType) {
     String helixJobQueueName = getHelixJobQueueName(taskType);
     LOGGER.info("Deleting task queue: {} for task type: {}", helixJobQueueName, taskType);
     _taskDriver.delete(helixJobQueueName);
   }
 
   /**
+   * Get all task queues.
+   *
+   * @return Set of task queue names
+   */
+  public synchronized Set<String> getTaskQueues() {
```

Comment:
By reading the Helix code, I don't think methods in TaskDriver are thread safe, so for now I made all of them synchronized.